src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpPreferFrameworkTypeDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.PreferFrameworkType;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpPreferFrameworkTypeDiagnosticAnalyzer : PreferFrameworkTypeDiagnosticAnalyzerBase<SyntaxKind>
```

Comment:
Can the base class be parameterized on the syntax node type you expect?  That way you don't need to define things in this class in terms of SyntaxNode?